src/algebra/ring/ulift.lean

```diff
@@ -104,8 +104,14 @@ by refine_struct { .. ulift.ring };
 tactic.pi_instance_derive_field
 
 instance field [field α] : field (ulift α) :=
-begin refine_struct { zero := (0 : ulift α), inv := has_inv.inv, div := has_div.div,
-  zpow := λ n a, ulift.up (a.down ^ n),
+begin
+  have of_rat_mk : ∀ a b h1 h2, ulift.up (field.rat_cast ⟨a, b, h1, h2⟩ : α) = ↑a * (↑b)⁻¹,
+  { intros a b h1 h2,
+    ext,
+    rw [field.rat_cast_mk, mul_down, inv_down, cast_nat_down, cast_int_down] },
+  refine_struct { zero := (0 : ulift α), inv := has_inv.inv, div := has_div.div,
+  zpow := λ n a, ulift.up (a.down ^ n), rat_cast := λ a, ulift.up (field.rat_cast a),
```

Comment:
Can you add a standalone `ulift.has_rat_cast` instance to match `ulift.has_nat_cast`? Along with the `ulift.rat_cast_down lemma`